platform/qt/src/bidi.cpp

```diff
@@ -0,0 +1,63 @@
+#include <memory>
+
+#include <mbgl/text/bidi.hpp>
+
+#include <QString>
+#include <QTextBoundaryFinder>
+
+namespace mbgl {
+
+class BiDiImpl {
+public:
+    QString string;
+};
+
+std::u16string applyArabicShaping(const std::u16string& input) {
+    QString utf16string = QString::fromStdU16String(input);
+    return utf16string.toStdU16String();
+}
+
+ProcessedBiDiText::ProcessedBiDiText(BiDi& p_bidi) : bidi(p_bidi) {
+}
+
+void ProcessedBiDiText::mergeParagraphLineBreaks(std::set<int32_t>& lineBreakPoints) {
+    QTextBoundaryFinder finder(QTextBoundaryFinder::Line, bidi.impl->string);
```

Comment:
Looking at the documentation for QTextBoundaryFinder::Line ("Finds possible positions for breaking the text into multiple lines."), it seems like this will insert a line break everywhere a line break is possible? Although maybe I'm misreading, because I expect that would have broken some tests...

The intention of this method is only to insert line breaks where the ICU BiDi algorithm absolutely requires them (for "block separators"), in case the calling code for whatever reason decided not to explicitly break at one of them. The common block separator is "\n", but there are more exotic ones like U+001C "INFORMATION SEPARATOR FOUR". For stubbing this code out on Qt, I don't think we need to implement any of this behavior (beyond maybe what's necessary to pass tests).